cpp/src/strings/replace/replace.cu

```diff
@@ -52,531 +49,388 @@ namespace detail {
- * @param chars_start Offset of the first character in the string column.
- * @param chars_end   Offset beyond the last character in the string column to search.
- * @param d_target    String to search for within the string column.
- * @param d_repl      Replacement string if target string is found.
- * @param maxrepl     Maximum times to replace if target appears multiple times in a string.
- * @param stream      CUDA stream to use for device operations
- * @param mr          Device memory resource used to allocate the returned column's device memory
- * @return New strings column.
+ * Performs the multi-replace operation with a thread per string.
+ * This performs best on smaller strings. @see AVG_CHAR_BYTES_THRESHOLD
  */
-std::unique_ptr<column> replace_char_parallel(strings_column_view const& strings,
-                                              size_type chars_start,
-                                              size_type chars_end,
-                                              string_view const& d_target,
-                                              string_view const& d_repl,
-                                              int32_t maxrepl,
-                                              rmm::cuda_stream_view stream,
-                                              rmm::mr::device_memory_resource* mr)
-{
-  auto const strings_count = strings.size();
-  auto const offset_count  = strings_count + 1;
-  auto const d_offsets   = strings.offsets().begin<int32_t>() + strings.offset();  // TODO: PR 14824
-  auto const d_in_chars  = strings.chars_begin(stream);
-  auto const chars_bytes = chars_end - chars_start;
-  auto const target_size = d_target.size_bytes();
-
-  // detect a target match at the specified byte position
-  device_span<char const> const d_chars_span(d_in_chars, chars_end);
-  auto target_detector = [d_chars_span, d_target] __device__(size_type char_idx) {
-    auto target_size = d_target.size_bytes();
-    auto target_ptr  = d_chars_span.begin() + char_idx;
-    return target_ptr + target_size <= d_chars_span.end() &&
-           d_target.compare(target_ptr, target_size) == 0;
-  };
-
-  // Count target string matches across all character positions, ignoring string boundaries and
-  // overlapping target strings. This may produce false-positives.
-  size_type target_count = thrust::count_if(rmm::exec_policy(stream),
-                                            thrust::make_counting_iterator<size_type>(chars_start),
-                                            thrust::make_counting_iterator<size_type>(chars_end),
-                                            target_detector);
-  if (target_count == 0) {
-    // nothing to replace, copy the input column
-    return std::make_unique<cudf::column>(strings.parent(), stream, mr);
-  }
+struct replace_fn {
+  column_device_view const d_strings;
+  string_view d_target;
+  string_view d_replacement;
+  cudf::size_type maxrepl;
+  int32_t* d_offsets{};
```

Comment:
Is this supposed to be size_type?

Reply:
Yep. Pretty old code that just got moved here.